wordpress-mobile/WordPress-FluxC-Android
# Woo/shipping label product ids fix

The GET shipping labels API endpoint now provides the `product_ids` field in the [response json](https://github.com/Automattic/woocommerce-services/pull/2125). This is needed in order to fetch the product details (images, name, SKU etc) associated with the product ids of a shipping label. This PR adds the `product_ids` field to the local db for use in the Woo app.

#### To test
- Open the example app and click on `Woo` -> `Shipping Labels` -> `Select site` -> `Fetch shipping labels`. Enter a valid order id and verify that the shipping labels for the order is fetched correctly and stored in the local db correctly. Check that the `WCShippingLabelModel` table in `Database Inspector` includes the `product_ids` field and it is not empty.
- Run `WCShippingLabelSqlUtilsTest` and `WCShippingLabelStoreTest`.
- The `product_ids` field is not available by default in all the API responses. For shipping labels created before the 1.24.1 version was released, the `product_ids` field will not be available. Since I had a few shipping labels created before then, I was able to test this scenario. This scenario is added to the `WCShippingLabelStoreTest` but if anyone would like to test this in the example app, please reach out to me so that I can provide you access to my test site :)
